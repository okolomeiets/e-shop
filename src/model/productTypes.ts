export interface Product {
  id: number
  title: string
  description: string
  price: number
  brand: string
  category: string
  discountPercentage?: number
  images?: string[]
  rating: number
  stock: number
  thumbnail: string
}
