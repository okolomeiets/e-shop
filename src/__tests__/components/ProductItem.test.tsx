import React from 'react';
import { render } from '@testing-library/react';
import ProductItem from '../../components/Product';
import { Product } from '../../model/productTypes';

const product: Product = {
  id: 1,
  title: 'Test Product',
  description: 'A description for the test product',
  price: 100,
  discountPercentage: 10,
  category: 'Electronics',
  brand: 'TestBrand',
  rating: 4.5,
  images: ['img1.jpg', 'img2.jpg'],
  stock: 20,
  thumbnail: 'thumb.jpg',
};

jest.mock('../../contexts/AdminContext', () => ({
  useAdmin: () => ({ isManagerToolsVisible: true }),
}));
jest.mock('../../contexts/ProductsContext', () => ({
  useProducts: () => ({ updateProduct: jest.fn() }),
}));

describe('ProductItem Component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<ProductItem product={product} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
