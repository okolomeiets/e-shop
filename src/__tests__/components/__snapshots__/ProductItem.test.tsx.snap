// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1l5bmg7-MuiPaper-root-MuiCard-root"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <button
        class="slick-arrow slick-prev slick-disabled"
        data-role="none"
        style="display: block;"
        type="button"
      >
         Previous
      </button>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Product Image 1"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1ofhrvj-MuiCardMedia-root"
                src="img1.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <img
                alt="Product Image 2"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1ofhrvj-MuiCardMedia-root"
                src="img2.jpg"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="slick-arrow slick-next"
        data-role="none"
        style="display: block;"
        type="button"
      >
         Next
      </button>
      <ul
        class="slick-dots"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
      </ul>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1v8nte1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
              data-testid="product-category"
              href="/"
            >
              Electronics
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-bvyohc-MuiTypography-root-MuiLink-root"
              data-testid="product-brand"
              href="/"
            >
              TestBrand
            </a>
          </li>
        </ol>
      </nav>
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1vco6p-MuiTypography-root"
        data-testid="product-title"
      >
        Test Product
      </div>
      <span
        aria-label="5 Stars"
        class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-w6b4hg-MuiRating-root"
        data-testid="product-rating"
        role="img"
      >
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 css-tb8f36-MuiTypography-root"
        data-testid="product-price"
      >
        Price: $100
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-1g4r38l-MuiTypography-root"
        data-testid="product-price-with-discount"
      >
        Price with Discount: $90.00
      </p>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-1lvdoz9-MuiTypography-root"
        data-testid="product-discount"
      >
        Save 10%!
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body2 css-16thyp8-MuiTypography-root"
        data-testid="product-description"
      >
        A description for the test product
      </p>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ManageAccountsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="10"
              cy="8"
              r="4"
            />
            <path
              d="M10.67 13.02c-.22-.01-.44-.02-.67-.02-2.42 0-4.68.67-6.61 1.82-.88.52-1.39 1.5-1.39 2.53V20h9.26c-.79-1.13-1.26-2.51-1.26-4 0-1.07.25-2.07.67-2.98M20.75 16c0-.22-.03-.42-.06-.63l1.14-1.01-1-1.73-1.45.49c-.32-.27-.68-.48-1.08-.63L18 11h-2l-.3 1.49c-.4.15-.76.36-1.08.63l-1.45-.49-1 1.73 1.14 1.01c-.03.21-.06.41-.06.63s.03.42.06.63l-1.14 1.01 1 1.73 1.45-.49c.32.27.68.48 1.08.63L16 21h2l.3-1.49c.4-.15.76-.36 1.08-.63l1.45.49 1-1.73-1.14-1.01c.03-.21.06-.41.06-.63M17 18c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
